### PR TITLE
Add support to use array in `controller` param in config

### PR DESCRIPTION
**Pull request by viktorruskai**
## Summary
- This PR contains a small feature where you can use an array in the config file. It will help your IDE navigate to the method you are using.

```php
return [
    'route' => [
        // The prefix for routes; do NOT use a leading slash!
        'prefix' => 'graphql',

        //  Normal format: 
        // 'controller' => GraphQLController::class . '@query',

        // Array format:
        'controller' => [GraphQLController::class, 'query'],
        ...

        'graphiql' => [
               'controller' => [GraphQLController::class, 'graphiql'],
               ...
        ],
        ...
```

- In your IDE it should look like this:
Screenshot 2022-03-04 at 15 46 34

---

Type of change:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Misc. change (internal, infrastructure, maintenance, etc.)

Checklist:
- [x] Existing tests have been adapted and/or new tests have been added
- [x] Add a CHANGELOG.md entry
- [x] Update the README.md
- [x] Code style has been fixed via `composer fix-style`

### CODE REVIEW

**Comment by viktorruskai** [comment]
@mfn I updated the Pull request.

File: README.md [hunk 1]
```diff
@@ -2655,7 +2655,7 @@ To prevent such scenarios, you can add the `UnusedVariablesMiddleware` to your
     The default makes the API available via `/graphql`
   - `controller`\
     Allows overriding the default controller class, in case you want to extend or
-    replace the existing one.
+    replace the existing one. (also supports `array` format)
```
Comment (mfn) [inline README.md:2658]:
```suggestion
    replace the existing one (also supports `array` format).
```

File: src/routes.php [hunk 2]
```diff
@@ -38,6 +38,10 @@ function (Router $router) use ($config, $routeConfig): void {
                     'middleware' => $schemaConfig['middleware'] ?? $routeConfig['middleware'] ?? null,
                 ]);
 
+                if (\is_array($actions['uses']) && isset($actions['uses'][0], $actions['uses'][1])) {
```
Comment (mfn) [inline src/routes.php:41]:
```suggestion
                // Support array syntax: `[Some::class, 'method']`
                if (\is_array($actions['uses']) && isset($actions['uses'][0], $actions['uses'][1])) {
```

File: src/routes.php [hunk 3]
```diff
@@ -69,6 +73,10 @@ function (Router $router) use ($config, $routeConfig): void {
             'middleware' => $graphiqlConfig['middleware'] ?? [],
         ],
         function (Router $router) use ($config, $graphiqlConfig): void {
+            if (\is_array($graphiqlConfig['controller']) && isset($graphiqlConfig['controller'][0], $graphiqlConfig['controller'][1])) {
```
Comment (mfn) [inline src/routes.php:76]:
```suggestion
            // Support array syntax: `[Some::class, 'method']`
            if (\is_array($graphiqlConfig['controller']) && isset($graphiqlConfig['controller'][0], $graphiqlConfig['controller'][1])) {
```

**Review by mfn — CHANGES_REQUESTED** [review]
LGTM!

Can you please:
- work in my feedback
- add an example for array syntax in the config itself, .e. g
  ```
          // The controller/method to use in GraphQL request.
          // Also supported array syntax: `[\Rebing\GraphQL\GraphQLController::class, 'query']`
          'controller' => \Rebing\GraphQL\GraphQLController::class . '@query',
  ```
- resolve the conflicts with master?

After that, I'm happy to merge the PR!

**Comment by viktorruskai** [comment]
@mfn I implemented everything you mentioned.

**Comment by mfn** [comment]
I guess we've to run `composer phpstan-baseline` to update the baseline; not sure what the appropriate code/phpdoc fix would be and I would be fine adding it to the baseline.

(note: I generated it locally but I can't push to your branch)

**Comment by viktorruskai** [comment]
@mfn Yeah, sure ...  I updated the baseline. Also, I tried to set proper PHPDoc with [PHPStan Array Shape](https://phpstan.org/writing-php-code/phpdoc-types#array-shapes), but unsuccessfully.

**Comment by mfn** [comment]
@viktorruskai thanks!

Unfortunately the baseline needs to be generated with the same PHP version running on GHA, which is 8.1 and not 7.4

I fixed this with https://github.com/rebing/graphql-laravel/pull/916 , which included your PR 🎉 

Thanks for your contribution!

**Pull request merged**